traitlets/config/configurable.py

```diff
@@ -203,8 +210,13 @@ def _config_changed(self, change):
         section_names = self.section_names()
         self._load_config(change.new, traits=traits, section_names=section_names)
 
-    def update_config(self, config):
-        """Update config and load the new values"""
+    def update_config(self, config, skip_env=False):
```

Comment:
I came to the conclusion that not respecting the environment, by default, 
is not the correct behavior for this feature.
Let me explain why:

Previous versions of traitlets had not any envvar capability.
Newer versions of traitlets would support that, but client-apps would NOT get that for free; they have to add `tag(envvar='FOO')` to specific traits.  Otherwise, they will continue to behave EXACTLY as before.

Now, having by default `respect_env=False` means that for clients, modifying their tags is not enough; they must also changes all `update_config()/load_config_file()` sites.  
For instance, in [my application](https://co2mpas.io) that's in 4 places + ~20 sites in test-cod.
But why?

That's the point of a new feature: 
it should come with all hassle-free defaults, predefined.
For this PR, it is to follow the priority given in the OP of #439.

That practical point is this:
- For old clients, nothing will change.
- For those wanting the new functionality, the current defaults makes more sense.